pennylane/gradients/parameter_shift.py

```diff
@@ -331,8 +331,22 @@ def var_param_shift(tape, argnum, shift=np.pi / 2, gradient_recipes=None, f0=Non
     def processing_fn(results):
         # We need to expand the dimensions of the variance mask,
         # and convert it to be the same type as the results.
-        mask = qml.math.convert_like(qml.math.reshape(var_mask, [-1, 1]), results[0])
-        f0 = qml.math.expand_dims(results[0], -1)
+        res = results[0]

```

Comment:
What's going on here, and why is only the parameter shift gradient affected?

Reply:
ohhhhh.

This is a bug uncovered by the new integration tests! 

Basically, the old approach didn't work for QNodes of the form `return qml.probs(), qml.expval()` :(